app/src/main/java/org/schabi/newpipe/fragments/MainFragment.java

```diff
@@ -136,16 +142,23 @@ public boolean onOptionsItemSelected(MenuItem item) {
     // Tabs
     //////////////////////////////////////////////////////////////////////////*/
 
-    public void setupTabs() {
+    private void setupTabs() {
         tabsList.clear();
         tabsList.addAll(tabsManager.getTabs());
 
+        boolean wasAdapterNull = pagerAdapter == null;
         if (pagerAdapter == null || !pagerAdapter.sameTabs(tabsList)) {
```

Comment:
I think the name was confusing at first and I decided to keep things clear, however, later on I renamed to something more readable and indeed, it would be better to do this.